api/envoy/api/v2/route/route.proto

```diff
@@ -349,6 +349,9 @@ message RouteMatch {
   message TlsContextMatchOptions {
     // If specified, the route will match against whether or not a certificate is presented.
     google.protobuf.BoolValue presented = 1;
+
+    // If specified, the route will match against whether or not a certificate is validated.
+    google.protobuf.BoolValue validated = 2;
```

Comment:
Ideally we make this an enum (and deprecate the old way of doing this).

Reply:
These fields are used for route matching, and are effectively a tribool - true/false/not-defined
 eg: for 'validated', by default we don't route based on validation status, unless specifically set to true / false, example snippet from previous PR #8248 :
```
"routes": [
  {
    "match": { "prefix": "/", "credentials": { "validated": false } },
    "route": {
      "cluster": "venue-edge-access-control",
      ...
    }
  },
  {
    "match": { "prefix": "/", "credentials": { "validated": true },
               "headers": [ { "name": ":authority", "prefix_match": "service-actual-target-service" }
               ]
    },
    "route": {
      "cluster": "service-actual-target-service"
    }
  }
...
```

Reply:
Ack. Is there a reason they are `BoolValue` other than history? What is the default value? If it's false, I suggest leaving a `[#next-major-version: change type to bool]` for both of these fields.

Reply:
The default is that it is not specified so will not be routed against. It is only routed against if specified in the configuration. This is why we've used a BoolValue (like a tribool). Is there another recommended way of defining this?